src/core_plugins/kbn_vislib_vis_types/public/controls/heatmap_options.js

```diff
@@ -0,0 +1,56 @@
+import uiModules from 'ui/modules';
+import heatmapOptionsTemplate from 'plugins/kbn_vislib_vis_types/controls/heatmap_options.html';
+import _ from 'lodash';
+const module = uiModules.get('kibana');
+
+module.directive('heatmapOptions', function ($parse, $compile, getAppState) {
```

Comment:
This comment isn't specific to this file; however, if I modify the colors in the series using the legend, and then define 'Custom Ranges' colors, it's causing some weird behaviors.